torchvision/prototype/transforms/functional/_misc.py

```diff
@@ -39,4 +39,5 @@ def gaussian_blur_image_tensor(
 
 
 def gaussian_blur_image_pil(img: PIL.Image, kernel_size: List[int], sigma: Optional[List[float]] = None) -> PIL.Image:
+    # It would really help to remove to_tensor from here vfdev-5
```

Comment:
I'd really like to help with that but there is no inverse for `pil_to_tensor` without rescaling the range...

Reply:
Related: https://github.com/pytorch/vision/pull/5500#discussion_r816525315

Reply:
@vfdev-5 I would like to avoid offering this method as part of the new API. Based on the implementation of `to_pil_image` I see that the x255 multiplication happens ONLY if the input is floating point. Hence I think we could avoid this by ensuring the image is of uint8 type prior using `to_pil_image`. Thoughts?

Reply:
@datumbox Let's say there are two cases we would like to cover (for instance without implementing `val_range`).
Input is PIL and its mode can be `"RGB"` or `"F"`.

```python
t_img = to_tensor(img)
# If RGB (uint8) -> [0-1], f32 tensor (rescaled by 255.0)
# If F (float32) -> f32 tensor (no rescale)

# t_img is float32 tensor
output = F_t.gaussian_blur(t_img, kernel_size, sigma)
# output is float32 tensor

output_pil = to_pil_image(output)
# No matter the input type output_pil is
# RGB uint8 Pillow image
# There is a bug here as we should pass origin mode
# It should be output_pil = to_pil_image(output, mode=img.mode)
```

> Hence I think we could avoid this by ensuring the image is of uint8 type prior using to_pil_image

If input is float32, conversion to uint8 may lead to precision reduction, I think.

Concerning the implementation, we can either make `to_tensor/to_pil_image` private and thus do not expose in the new API. Another solution could be to code PIL->Tensor->PIL explicitly:

```
import torch

def foo(img):
    t_img = torch.from_numpy(np.array(img))
    do_squeeze = False
    if t_img.ndim == 2:
        do_squeeze = True
        t_img = t_img.unsqueeze(-1)
    t_img = t_img.permute(2, 0, 1).contiguous()
    t_img = t_img.to(torch.float32)

    t_out = F.gaussian_blur(t_img, kernel_size=(3, 3), sigma=(0.5, 0.5))

    np_out = t_out.cpu().numpy()
    np_out = np.transpose(np_out, (1, 2, 0))
    if do_squeeze:
        np_out = np_out[:, :, 0]
    return PIL.Image.fromarray(np_out, mode=img.mode)
```

Reply:
@vfdev-5 What I had in mind was to do:
```python
t_img = F.pil_to_tensor(img)
output = F.gaussian_blur(t_img, (3, 3), [1.0, 1.0])
output_pil = to_pil_image(output)
```

An alternative if we are afraid of loss of precision could be:
```python
t_img = F.pil_to_tensor(img).div(255.0)
output = F.gaussian_blur(t_img, (3, 3), [1.0, 1.0])
output_pil = to_pil_image(output)
```

I just don't want to use `to_tensor` on the new API. Ideally we should deprecate and throw warnings if someone uses it.

Reply:
Addressed in #5503.